tests/kernel/mem_protect/futex/src/main.c

```diff
@@ -452,6 +451,81 @@ void test_user_futex_bad(void)
 	zassert_equal(ret, -ETIMEDOUT, "didn't time out");
 }
 
+void futex_wait_wake(void *p1, void *p2, void *p3)
+{
+	int32_t ret_value;
+
+	/* Test user thread can make wait without error
+	 * Use assertion to verify k_futex_wait() returns 0
+	 */
+	ret_value = k_futex_wait(&simple_futex, 13, K_FOREVER);
+	zassert_equal(ret_value, 0, NULL);
+
+	/* Test user thread can make wake without error
+	 * Use assertion to verify k_futex_wake() returns 1,
+	 * because only 1 thread wakes
+	 */
+	ret_value = k_futex_wake(&simple_futex, false);
+	zassert_equal(ret_value, 1, NULL);
+}
+
+void futex_wake(void *p1, void *p2, void *p3)
+{
+	int32_t atomic_ret_val;
+
+	k_futex_wake(&simple_futex, false);
+
+	k_futex_wait(&simple_futex, 13, K_FOREVER);
+
+	/* Test user can write to the futex value
+	 * Use assertion to verify substraction correctness
+	 * Initial value was 13, after atomic_sub() must be 12
+	 */
+	atomic_sub(&simple_futex.val, 1);
+	atomic_ret_val = atomic_get(&simple_futex.val);
+	zassert_equal(atomic_ret_val, 12, NULL);
+}
+
+/**
+ * @brief Test kernel supports locating kernel objects without private kernel
+ * data anywhere in memory, control access with the memory domain configuration
+ *
+ * @details For that test kernel object which doesn't contain private kernel
+ * data will be used futex. Test performs handshaking between two user threads
+ * to test next requirements:
+ * - Place a futex simple_futex in user memory using ZTEST_BMEM
+ * - Show that user threads can write to futex value
+ * - Show that user threads can make wait/wake syscalls on it.
+ *
+ * @see atomic_set(), atomic_sub(), k_futex_wake(), k_futex_wait()
+ *
+ * @ingroup kernel_futex_tests
+ */
+void test_futex_locate_access(void)
+{
+
+	atomic_set(&simple_futex.val, 13);
+
+	k_thread_create(&futex_tid, stack_1, STACK_SIZE,
```

Comment:
why are you doing this in a child thread?
couldn't this test have been defined using ztest_user_unit_test?

Reply:
I did changes in pushed new commit that runs that test using ztest_user_unit_test

Reply:
well I think you don't need a child thread, unless im missing something it doesn't seem like it is synchronizing with any other threads, it just runs and does its own thing. can't we move this logic up to the body of this function, and run it in user mode?